pkg/common/utils.go

```diff
@@ -695,3 +698,35 @@ func NewLimiter(limit, burst int, emptyBucket bool) *rate.Limiter {
 
 	return limiter
 }
+
+// splitWithEscape splits a string with the given delimiter and escape character.
+func splitWithEscape(s string, delim rune, escape rune) []string {
+	var parts []string
+	var buf bytes.Buffer
+	escaped := false
+
+	for _, r := range s {
+		switch {
+		case escaped:
+			if r == delim {
```

Comment:
Why can't we just do
  case escaped:
   if r != escape {
    buf.WriteRune(r)
    escaped = false
  }

I think the current code will do the wrong thing on foo\,,bar.

also I think both versions of the code do the wrong thing on foo\\,bar, usually that would escape the backslash and not escape the comma, but here this is turned into an escaped comma?

Reply:
@mattcary You're right, it can be simplified as suggested. As for the behavior of escaping, what do you think the correct behavior should be? Or which unit test is wrong specifically? 

In case of `foo,,bar` input `splitWithEscape` will output `"foo", "", "bar"` which will later fail validation as it's not a valid format for a tag here: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/afdbb016571a0ad3104489681b5edb2a426c8bd3/pkg/common/utils.go#L322 


> here this is turned into an escaped comma?

I think you could say that, it's actually the intention of this change to allow users escaping the commas somehow so they can be used in tag value (as documented). Maybe there's a better solution?